ibis-server/app/mdl/rewriter.py

```diff
@@ -46,17 +45,17 @@ def rewrite(self, sql: str) -> str:
                 },
                 content=orjson.dumps({"manifestStr": self.manifest_str, "sql": sql}),
             )
-            rewritten_sql = (
-                r.text if r.status_code == httpx.codes.OK else r.raise_for_status()
-            )
-            logger.debug("Rewritten SQL: %s", rewritten_sql)
+            rewritten_sql = r.raise_for_status().text
+            logger.debug("Rewritten SQL: {}", rewritten_sql)
             return (
                 rewritten_sql
                 if self.data_source is None
                 else self.transpile(rewritten_sql)
             )
         except httpx.ConnectError as e:
             raise ConnectionError(f"Can not connect to Wren Engine: {e}")
+        except httpx.HTTPStatusError as e:
+            raise RewriteError(e.response.text)
```

Comment:
I think we can also wrap the rewrite exception into RewriteError for EmbeddedEngineRewriter, then I guess we can be powered by the exception handler in `v3/connector.py` to generate the response.